client/state/sites/selectors.js

```diff
@@ -741,6 +741,23 @@ export function hasJetpackSiteJetpackThemes( state, siteId ) {
 }
 
 /**
+ * Determines if the Jetpack plugin of a Jetpack Site has extend themes management features.
+ * Returns null if the site is not known or is not a Jetpack site.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} true if the site has Jetpack extended themes management features
+ */
+export function hasJetpackSiteJetpackThemesExtendedFeatures( state, siteId ) {
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
```

Comment:
Yeah, this touches on a broader discussion that we recently had on the calypso framework channel on Slack. Currently, this is sort of an established pattern found in other Jetpack related selectors, too. I agree it makes sense to challenge this pattern -- I think it'd even make sense to drop the `isJetpackSite` check inside these individual selectors altogether and require consumers to take care of that -- but doing this inside this one particular selector would break the pattern. So I'll keep this for now. (If I had more time, I'd file a PR that removes those `isJetpackSite` checks as a basis for further discussion...)